### PR TITLE
Feature/sgw env variables

### PR DESCRIPTION
**Pull request by sridevi-15**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- SGW file management test cases
> Test cases for substituting variables with environment variables
> Test cases for local jscode  file 
> Test cases for hosting jscode remotely

Jenkins job : http://uberjenkins.sc.couchbase.com:8080/job/cen7-sync-gateway-functional-tests-xattrs-noConflicts-serverSsl/174/testReport/

### CODE REVIEW

File: testsuites/syncgateway/functional/tests/test_webhooks.py [hunk 1]
```diff
@@ -310,6 +314,237 @@ def test_webhooks_crud(params_from_base_test_setup, sg_conf_name, filtered):
...
+
+    if sync_gateway_version < "3.0.0":
+        pytest.skip("this feature not available below 3.0.0")
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+    cluster_helper = ClusterKeywords(cluster_config)
+    cluster_hosts = cluster_helper.get_cluster_topology(cluster_config)
+    sg_admin_url = cluster_hosts["sync_gateways"][0]["admin"]
+    sg_url = cluster_hosts["sync_gateways"][0]["public"]
+    username = "autotest"
+    password = "password"
```
Comment (mgcouchbase) [inline testsuites/syncgateway/functional/tests/test_webhooks.py:449]: It seems like this test do not required reset sg, can we avoid to save some execution time . (Change the username & password  and have the unique session this test , when you decided to skip)

**Review by mgcouchbase — APPROVED** [review]
Please take a look the comments

File: testsuites/syncgateway/functional/tests/test_webhooks.py [hunk 2]
```diff
@@ -310,6 +314,237 @@ def test_webhooks_crud(params_from_base_test_setup, sg_conf_name, filtered):
...
+@pytest.mark.oscertify
+def test_webhook_filter_external_js(params_from_base_test_setup, setup_webserver):
+    """
+    "1. Create valid js function  for import filter and host it in local machine to access jscode with http url
+    webhook filter : add doc.type if doc has data with webhook filter value, otherwise add doc.type is ignore
+    2. Create sgw config and point the webhook filter jsfunction to the webhook function hosted in the localmachine and can be accessed by http url
+    Start sync gateway
+    3. Verify SGW starts sucessfully
+    4. create docs in SDK few docs with data as webhook_filter and some without non_webhook_filter data
+    5. Verfiy webhook events generated and the docs with webhook filter should have in in the events
```
[the remark this answers is not in the document]
Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_webhooks.py:328]: could you point me the word which has typo?

File: utilities/copy_files_to_nodes.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import os
+from keywords.constants import ENVIRONMENT_FILE
+from libraries.provision.ansible_runner import AnsibleRunner
+
+
+def create_files_with_content(content, sg_platform, sg_hostname, file_name, cluster_conf, path=None):
+    ansible_runner = AnsibleRunner(cluster_conf)
+    if path is None:
+        path = "/tmp/" + file_name
```
Comment (euniceatcouchbase) [inline utilities/copy_files_to_nodes.py:9]: does /tmp directory always exist? and if you need copying files to this directory, you will need to clean up, while you may have already done that in ansible script. 
is it possible to use platform tmp directory? such as every platform has it own tmp location and the OS is in charge of cleaning them up. 
just a discussion, not requesting any code change.

File: testsuites/syncgateway/functional/tests/test_webhooks.py [hunk 4]
```diff
@@ -310,6 +314,237 @@ def test_webhooks_crud(params_from_base_test_setup, sg_conf_name, filtered):
...
+    yield{
+        "webhook_server": webhook_server
+    }
+
+    webhook_server.stop()
+    process.kill()
+
+
+@pytest.fixture(scope="function")
+def setup_webserver_jssslon():
```
Comment (euniceatcouchbase) [inline testsuites/syncgateway/functional/tests/test_webhooks.py:538]: i am kind of guessing what this function name means, if you don't mind, setup_webserver_js_ssl_on() may be more straightforward. :)

**Review by euniceatcouchbase — APPROVED** [review]
approved with some comments

**Review by euniceatcouchbase — APPROVED** [review]
super 👍

File: testsuites/syncgateway/functional/tests/test_webhooks.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mgcouchbase) [reply testsuites/syncgateway/functional/tests/test_webhooks.py:328]: Verfiy*** webhook

**Pull request merged**